lib/Exception/UserTokenException.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * @author Sujith Haridasan <sharidasan@owncloud.com>
+ *
+ * @copyright Copyright (c) 2018, ownCloud GmbH
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\UserManagement\Exception;
+
+use Throwable;
+
+class UserTokenException extends \Exception {
+	const TOKEN_INVALID = 1;
+	const TOKEN_EXPIRED = 2;
+	const TOKEN_MISMACTH = 3;
+
+	/** @var int */
+	private $tokenFailure;
+
+	public function __construct($exceptionMessageType, $code = 0, Throwable $previous = null) {
+		$message = '';
+		$l10n = \OC::$server->getL10N('user_management');
+		if ($exceptionMessageType === self::TOKEN_INVALID) {
+			$message = $l10n->t('Could not set password because the token is invalid');
+			$this->tokenFailure = self::TOKEN_INVALID;
```

Comment:
Use the exception code instead of this new attribute?

Reply:
Removed the method `getTokeFailureType` and assigned `code` with value of `exceptionMessageType`. And in the UsersController, `getCode` is used.